src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployerProperties.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.kubernetes;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author Florian Rosenberg
+ * @author Thomas Risberg
+ */
+@ConfigurationProperties(prefix = "deployer.kubernetes")
+public class KubernetesAppDeployerProperties {
+
+	/**
+	 * Secrets for a access a private registry to pull images.
+	 */
+	private String imagePullSecret;
+
+	/**
+	 * Delay in seconds when the Kubernetes liveness check of the app container
+	 * should start checking its health status.
+	 */
+	// See http://kubernetes.io/v1.0/docs/user-guide/production-pods.html#liveness-and-readiness-probes-aka-health-checks}
+	private int livenessProbeDelay = 30;
+
+	/**
+	 * Timeout in seconds for the Kubernetes liveness check of the app container.
+	 * If the health check takes longer than this value to return it is assumed as 'unavailable'.
+	 */
+	// see http://kubernetes.io/v1.0/docs/user-guide/production-pods.html#liveness-and-readiness-probes-aka-health-checks}
+	private int livenessProbeTimeout = 2;
+
+	/**
+	 * Delay in seconds when the readiness check of the app container
+	 * should start checking if the module is fully up and running.
+	 */
+	// see http://kubernetes.io/v1.0/docs/user-guide/production-pods.html#liveness-and-readiness-probes-aka-health-checks}
+	private int readinessProbeDelay = 20;
+
+	/**
+	 * Timeout in seconds that the app container has to respond to its
+	 * health status during the readiness check.
+	 */
+	// see http://kubernetes.io/v1.0/docs/user-guide/production-pods.html#liveness-and-readiness-probes-aka-health-checks}
+	private int readinessProbeTimeout = 2;
+
+	/**
+	 * Memory to allocate for a Pod.
+	 */
+	private String memory = "512Mi";
+
+	/**
+	 * CPU to allocate for a Pod.
+	 */
+	private String cpu = "500m";
+
+	/**
+	 * Maximum allowed restarts for app that fails due to an error or excessive resource use.
+	 */
+	private int maxTerminatedErrorRestarts = 2;
+
+	/**
+	 * Maximum allowed restarts for app that is in a CrashLoopBackOff.
+	 */
+	private int maxCrashLoopBackOffRestarts = 4;
+
+//TODO: maybe?
```

Comment:
we can remove this comment.... I believe it would live at the boot level if we incorporated this functionality.